clients/edgex-foundry/device_client.go

```diff
@@ -54,21 +54,21 @@ func NewEdgexDeviceClient(coreMetaAddr, coreCommandAddr string) *EdgexDeviceClie
 
 // Create function sends a POST request to EdgeX to add a new device
 func (efc *EdgexDeviceClient) Create(ctx context.Context, device *devicev1alpha1.Device, options edgeCli.CreateOptions) (*devicev1alpha1.Device, error) {
-	dp := toEdgeXDevice(device)
-	klog.V(5).Infof("will add the Devices: %s", dp.Name)
-	dpJson, err := json.Marshal(&dp)
+	devs := []*devicev1alpha1.Device{device}
+	req := makeEdgeXDeviceRequest(devs)
+	klog.V(5).Infof("will add the Device: %s", device.Name)
+	reqBody, err := json.Marshal(req)
 	if err != nil {
 		return nil, err
 	}
 	postPath := fmt.Sprintf("http://%s%s", efc.CoreMetaAddr, DevicePath)
 	resp, err := efc.R().
-		SetBody(dpJson).Post(postPath)
+		SetBody(reqBody).Post(postPath)
 	if err != nil {
 		return nil, err
 	} else if resp.StatusCode() != http.StatusOK {
```

Comment:
In edgex 2.1 version, since it supports the creation of multiple devices at one time, the returned message is also different from 1.x. 
- First, after the request is sent successfully, the returned message status code is 207, indicates a multi-part response supportive of accepting multiple requests at once. 
- Second, the 'statusCode' property of each response in the returned array will indicate success or failure. 

For details, please refer to: https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.1.0#/default/post_device